image_data_guide/04b_tensorflow_training.ipynb

```diff
@@ -0,0 +1,889 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# TensorFlow Training with SageMaker"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Download | Structure | Preprocess | **Train Model**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<pre>\n",
+    "</pre>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Overview\n",
+    "* #### Dependencies\n",
+    "* #### Algorithm Hyperparameters\n",
+    "* #### Review the Training Script\n",
+    "* #### Estimator Configuration\n",
+    "* #### Training in Local Mode and Debugging\n",
+    "* #### Training on EC2 Instances\n",
+    "* #### Load Trained Model and Predict on Test Data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<pre>\n",
+    "</pre>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Dependencies\n",
+    "___\n",
+    "For this guide we'll use the SageMaker Python SDK version 2.9.2. By default, SageMaker Notebooks come with version 1.72.0. Other guides provided by Amazon may be set up to work with other versions of the Python SDK so you may wish to roll-back to 1.72.0.\n",
+    "\n",
+    "We will also debug our code by training on the instance running this notebook (Local Mode). In order to run through one epoch of training in a reasonable amount of time I advise using a notebook backed by a p2.xlarge instance. Once youre script has completely run locally and all bugs have been ironed out, then you can switch back to a smaller instance."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Update SageMaker Python SDK and TensorFlow"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[31mERROR: tensorflow 2.3.0 has requirement numpy<1.19.0,>=1.16.0, but you'll have numpy 1.19.1 which is incompatible.\u001b[0m\n",
+      "\u001b[33mWARNING: You are using pip version 20.0.2; however, version 20.2.3 is available.\n",
+      "You should consider upgrading via the '/home/ec2-user/anaconda3/envs/tensorflow2_p36/bin/python -m pip install --upgrade pip' command.\u001b[0m\n"
+     ]
+    }
+   ],
+   "source": [
+    "import sys\n",
+    "original_sagemaker_version = !conda list | grep -E \"sagemaker\\s\" | awk '{print $2}'\n",
+    "original_tensorflow_version = !conda list | grep -E \"tensorflow\\s\" | awk '{print $2}'\n",
+    "!{sys.executable} -m pip install -q \"sagemaker==2.9.2\" \"tensorflow-serving-api==2.3.0\" \"tensorflow==2.3.0\" \"tensorflow-datasets\""
```

Comment:
You can use %pip instead of sys.executable
Also I tried this with only installing the datasets package and it works fine. (no need to install the rest)

Reply:
just keeping he installation of tensorflow-datasets